SingularityService/src/test/java/com/hubspot/singularity/scheduler/SingularityMachinesTest.java

```diff
@@ -117,6 +116,74 @@ public void testDeadSlavesArePurged() {
     Assertions.assertEquals(0, agentManager.getObjectsFiltered(MachineState.DEAD).size());
   }
 
+  @Test
+  public void testMissingSlavesArePurged() {
```

Comment:
The addition of this test causes `testDeadSlavesArePurged` to fail when run after it but I'm not clear on why

Reply:
Decent chance it's because the agent manager is shared between the two tests, so the overlapping agent ids + reconciliations cause the failures.

Could try changing the ids for this test to not overlap, or else try refactoring this test class to always use a fresh agent manager for each test.

Reply:
If you look into the base tests there is some `@After` annotated code that should be clearing out the in-memory zk instance used for testing after each test. The manager itself does not hold data, it just accesses it. The data is held in zk, which for tests is an in-memory curator. +1 to changing IDs to make it not overlap though still, since these bugs are often hard to track down/not worht the time if something can fix itmore easily

Reply:
I did actually try changing the ids such that they don't overlap and the failure was still occurring :\ I'll try refreshing the agent manager. Thanks Will

Reply:
The failure persists after changing the agent ids to not overlap between tests. I see the `@AfterEach` code in the base tests that clears out the leader cache and invalidates all in the zk task and deploy caches, though...will continue investigating

Reply:
Think I know where this might be. That test that now fails sets `configuration.setDeleteDeadAgentsAfterHours(1);` in the middle of the test. The SingularityConfiguration object is not currently reset between runs of tests that I can see. So it's possible that setting is remaining in effect and messing up subsequent tests. Might be able to put either something in the AfterEach to reset it, or just a finally block in your own test that will put it back to the previous default if you are changing it

Reply:
Thanks Steve, `configuration.deleteDeadAgentsAfterHours` is in fact not being reset between tests so resetting it fixed the problem